### PR TITLE
Encode non-resource types

### PR DESCRIPTION
**Pull request by jamesagnew**

### CODE REVIEW

File: hapi-fhir-structures-r4/src/test/java/ca/uhn/fhir/parser/RDFParserR4Test.java [hunk 1]
```diff
@@ -23,269 +27,62 @@
 import static org.hamcrest.Matchers.containsString;
 import static org.junit.jupiter.api.Assertions.assertEquals;
 import static org.junit.jupiter.api.Assertions.assertNotEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
 
-@Disabled
 public class RDFParserR4Test {
-	private static final Logger ourLog = LoggerFactory.getLogger(RDFParserR4Test.class);
-	private static FhirContext ourCtx = FhirContext.forR4();
+	private static final FhirContext ourCtx = FhirContext.forR4Cached();
```
Comment (jmarchionatto) [inline hapi-fhir-structures-r4/src/test/java/ca/uhn/fhir/parser/RDFParserR4Test.java:33]: Why were existing tests removed?

**Review by jmarchionatto — CHANGES_REQUESTED** [review]
Added a couple of questions

File: hapi-fhir-base/src/main/java/ca/uhn/fhir/parser/JsonParser.java [hunk 2]
```diff
@@ -1446,9 +1455,7 @@ private void managePrimitiveExtension(final IBase theValue, final RuntimeResourc
 				// Undeclared extensions
 				extractUndeclaredExtensions(theValue, extensions, modifierExtensions, myParent, null, theEncodeContext, theContainedResource);
 				// Declared extensions
-				if (def != null) {
-					extractDeclaredExtensions(theValue, def, extensions, modifierExtensions, myParent);
-				}
+				extractDeclaredExtensions(theValue, def, extensions, modifierExtensions, myParent);
```
[the remark this answers is not in the document]
Comment (jamesagnew) [reply hapi-fhir-base/src/main/java/ca/uhn/fhir/parser/JsonParser.java:1458]: `def` can't actually be null here, it's dereferenced 2 lines up. I did a scan and there is no way someone could call this method with that parameter being null either.

[the remark this answers is not in the document]
Comment (jmarchionatto) [reply hapi-fhir-base/src/main/java/ca/uhn/fhir/parser/JsonParser.java:1458]: Fair enough

**Review by jmarchionatto — APPROVED** [review]

**Pull request merged**